server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRefreshAction.java

```diff
@@ -62,7 +62,7 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
             @Override
             public RestResponse buildResponse(RefreshResponse response, XContentBuilder builder) throws Exception {
                 builder.startObject();
-                buildBroadcastShardsHeader(builder, request, response);
+                response.toXContent(builder, request);
                 builder.endObject();
                 return new BytesRestResponse(response.getStatus(), builder);
```

Comment:
you can shorten this by using `RestToXContentListener` instead of `RestBuilderListener`. If a response implements `ToXContent`, we don't need to do anything special.

Reply:
I see that for `RestToXContentListener`, its generic type should be a `ToXContentObject` but our `BroadcastResponse` is currently a `ToXContentFragment`. If we change it to `ToXContentObject`, we also need to modify all the subclasses of it. So maybe here we just keep it like this for this moment ?

Reply:
right, we can do that as a followup